caps/eth.md

```diff
@@ -105,25 +105,185 @@ of it (either because it was previously sent or because it was informed from thi
 originally). This is usually achieved by remembering a set of transaction hashes recently
 relayed by the peer.
 
-Transactions must be validated before re-propagating them. Relaying an invalid transaction
-results in peer disconnection.
+### Transaction Encoding and Validity
+
+Transaction objects exchanged by peers have one of two encodings. In definitions across
+this specification, we refer to transactions of either encoding using the identifier `txₙ`.
+
+    tx = {legacy-tx, typed-tx}
+
+Untyped, legacy transactions are given as an RLP list.
+
+    legacy-tx = [
+        nonce: P,
+        gas-price: P,
+        gas-limit: P,
+        recipient: {B_0, B_20},
+        value: P,
+        data: B,
+        V: P,
+        R: P,
+        S: P,
+    ]
+
+[EIP-2718] typed transactions are encoded as RLP byte arrays where the first byte is the
+transaction type (`tx-type`) and the remaining bytes are opaque type-specific data.
+
+    typed-tx = rlp(tx-type || tx-data)
```

Comment:
I mentioned this is Discord, but wanted to mention it again here: This is currently incorrect.

If I were implementing this from the specification (without any prior knowledge of what is correct) I would have code that looks something like:
```ts
const tx1 = [nonce, gas_price, gas_limit, recipient, value, data, V, R, S]
const tx2Data = getTxDataAsByteArray(...)
const tx2 = rlpEncode([type, ...tx2Data])
const payload = rlpEncode([tx1, tx2])
```
This would be incorrect as I double encoded tx2.  Currently, this specification is *telling* me to double encode tx2 by saying that a typed transaction is an `rlp(tx-type || tx-data)`, which is not what it actually is.

I may actually incorrectly rlp encode `tx1` as well, since it says it should be an `RLP list`, though I could see myself going either way on that (an RLP *encoded* list or an RLP *compatible* list).

Reply:
I've been thinking about this a lot last couple days, and want to leave it as-is in the spec. The `rlp(...)` construction is the only way we have to formally describe that the typed transaction is a RLP byte array. You have to see it in context:

```
tx = {legacy-tx, typed-tx}
legacy-tx = [...]
typed-tx = rlp(...)
```

I don't think that anyone will be confused by this, and even if they are, it would be noticed quickly when testing the implementation using the compliance test suite or against another implementation.

Reply:
We decided to remove the `rlp(...)` after all